app/models/user.rb

```diff
@@ -211,7 +211,11 @@ def grant_application_permissions(application, supported_permission_names)
   end
 
   def grant_permission(supported_permission)
-    application_permissions.where(supported_permission_id: supported_permission.id).first_or_create!
+    if persisted?
+      application_permissions.where(supported_permission_id: supported_permission.id).first_or_create!
+    else
+      supported_permissions << supported_permission
+    end
```

Comment:
I think there's a potential problem here because it's possible to add the same supported permission multiple times to an unsaved user. Saving that user will then fail with `ActiveRecord::RecordInvalid: Validation failed: Supported permission has already been taken`. I wonder if it's better to use something like:

`supported_permissions << supported_permission unless supported_permissions.include?(supported_permission)`

Which works in both the persisted and unpersisted case.